shared_model/interfaces/query_responses/signatories_response.hpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_SIGNATORIES_RESPONSE_HPP
+#define IROHA_SHARED_MODEL_SIGNATORIES_RESPONSE_HPP
+
+#include <boost/assert.hpp>
+#include "crypto/hash.hpp"
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/primitive.hpp"
+#include "interfaces/visitor_apply_for_all.hpp"
+#include "model/queries/responses/signatories_response.hpp"
+#include "utils/string_builder.hpp"
+
+namespace shared_model {
+  namespace interface {
+    /**
+     * Container of asset, for fetching data.
+     */
+    class SignatoriesResponse
+        : public Primitive<SignatoriesResponse,
+                           iroha::model::SignatoriesResponse> {
+     public:
+      /**
+       * @return All public keys attached to account
+       */
+      virtual types::PublicKeyCollectionType keys() const = 0;
+
+      /**
+       * Stringify the data.
+       * @return string representation of data.
+       */
+      std::string toString() const override {
+        return detail::PrettyStringBuilder()
+            .init("SignatoriesResponse")
+            .append("signatories")
+            .appendAll(keys(), [](auto &key) { return key->toString(); })
+            .finalize();
+      }
+
+      /**
+       * @return true if the data are same.
+       */
+      bool operator==(const ModelType &rhs) const override {
+        return keys() == rhs.keys();
+      }
+
+      /**
+       * Makes old model.
+       * @return An allocated old model of signatories response.
+       */
+      OldModelType *makeOldModel() const override {
+        OldModelType *oldModel = new OldModelType();
+        const auto vs = keys();
+        std::for_each(vs.begin(), vs.end(), [&oldModel](const auto &key) {
+          oldModel->keys.push_back(
+              key->template makeOldModel<iroha::pubkey_t>());
```

Comment:
> What is template here mean?
To compile success.
> error: use 'template' keyword to treat 'makeOldModel' as a dependent template name

Reply:
> Also, I feel here memory leak because you create an object on the heap and after creates a copy in the vector. Use emplace_back instead.

`makeOldModel<T>()` returns not pointer but object. And push_back() is same as emplace_back() if only one argument is used.

Reply:
Yes, I was mistaken with memory leak here. But what is the reason to create blob and make a redundant copy of it? I guess, better to use `emplace_back`.

Reply:
I see. `push_back` causes a redundant copy or move.